vectorbendertransformers.py

```diff
@@ -26,6 +26,83 @@ def __init__(self, pairsLayer, restrictToSelection):
     def map(self, p):
         return p
 
+# we introduce the affine transformation (by francesco marucci)
+class AffineTransformerFRA(Transformer):
+    def __init__(self, pairsLayer ):
+        Transformer.__init__(self, pairsLayer )      
+
+        # Make sure data is valid
+        assert len(self.pointsA)>=3
+        assert len(self.pointsA)==len(self.pointsB)
+        
+        self.affineParams=self.calculateAffineParams(self.pointsA,self.pointsB)
+        
+    def map(self, p):
+      
+        a=self.affineParams[0]
+        b=self.affineParams[1]
+        c=self.affineParams[2]
+        d=self.affineParams[3]
+        tx=self.affineParams[4]
+        ty=self.affineParams[5]
+        
+        return QgsPoint(a*Decimal(p[0])+b*Decimal(p[1])+tx, c*Decimal(p[0])+d*Decimal(p[1])+ty)
+
+    def calculateAffineParams(self, p1, p2 ):
```

Comment:
I split that function because I want to calculate the affine tranformation errors "before" to run the transformation, in ohers words I want to execute the "calculateAffineParameters" when the user choose a pair layer, as I made in my plugin version:
![image](https://cloud.githubusercontent.com/assets/1571351/26753946/52965a2e-4871-11e7-8693-c87cf7beeba0.png)




Yes you are rigth, I have to comment a bit the code.
in any case, I just translated the code from GRASS vector transform function:
https://grass.osgeo.org/programming6/transform_8c_source.html
maybe we have to do the attribution to GRASS...